racket-src/body/type-check-goal.rkt

```diff
@@ -135,9 +135,56 @@
                            (&& ((Ty_op <= Ty_field) ...)))
    ]
 
+  [; type-check ReifyFnPointer casts
+   (type-of/Operand Γ Operand Ty_fn)
+   (fn-ty-signature Γ Ty_fn (∀ KindedVarIds (implies Biformulas ((Ty_arg ...) -> Ty_ret))))
+   (where/error number_args ,(length (term (Ty_arg ...))))
+   (where/error Ty_fnptr (rigid-ty (fn-ptr "Rust" number_args) (Ty_arg ... Ty_ret)))
+   ----------------------------------------
+   (type-check-goal/Rvalue Γ
+                           (cast Operand as Ty_target)
+                           (∀ KindedVarIds (implies Biformulas (Ty_fnptr <= Ty_target))))
+   ]
+
   [----------------------------------------
    (type-check-goal/Rvalue Γ _ true-goal)
    ]
 
   )
 
+(define-judgment-form
+  formality-body
+  #:mode (fn-ty-signature I I O)
+  #:contract (fn-ty-signature Γ Ty Signature)
+
+  [(fn-ty-signature Γ Ty (∀ () (implies Biformulas (Tys_args -> Ty_ret))))
+   ----------------------------------------
+   (fn-ty-signature Γ
+                    (∀ KindedVarIds Ty)
+                    (∀ KindedVarIds (implies Biformulas (Tys_args -> Ty_ret))))
+   ]
+
+  [(fn-ty-signature Γ Ty (∀ () (implies () (Tys_args -> Ty_ret))))
```

Comment:
Same here, we would ideally get back a `(ForAll KindedVarIds (implies Biformulas_1 (...))` and then concatenate that with the biformulas we "peeled off"